server.py

```diff
@@ -59,10 +59,9 @@ def receipt():
 def ack(ru_ref, exercise_sid):
 
     if len(ru_ref) != 11 or len(exercise_sid) != 11:
-        logger.warning("Got a invalid ru_ref/exercise_sid".format(ru_ref=ru_ref, exercise_sid=exercise_sid))
+        logger.warning("Got an invalid ru_ref/exercise_sid", ru_ref=ru_ref, exercise_sid=exercise_sid)
```

Comment:
We should look at using `structlog` that way you can do `logger.bind(ru_ref=ru_ref, exercise_sid=exercise_sid)` so you dont have to specify on multiple log messages.